### PR TITLE
For #7425,Fix namespace auth failed problem

### PR DESCRIPTION
**Pull request by zrfan2016**
Please do not create a Pull Request without creating an issue first.

For #7425

## What is the purpose of the change

Hide the namespaces for user who has no access in the page header tabs.

## Brief changelog

1.NacosUser中新增Set<String> roles属性
2.登录中，将角色信息填充到当前登录用户信息中
3.NamespaceController获取当前登录用户信息
4.未开启校验或管理员账号查看所有namespace，否则根据用户权限查询相应namespace

## Verifying this change

1.配置文件中启用数据库mysql配置
2.管理员账号登录控制台，添加namespace、用户、角色、权限等。
3.退出管理员账号，使用添加账号登录查看

Follow this checklist to help us incorporate your contribution quickly and easily:

* [ ] Make sure there is a Github issue filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
* [ ] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
* [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
* [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/alibaba/nacos/tree/master/test).
* [ ] Run `mvn -B clean package apache-rat:check findbugs:findbugs -Dmaven.test.skip=true` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.

### CODE REVIEW

File: config/src/main/java/com/alibaba/nacos/config/server/service/repository/PersistService.java [hunk 1]
```diff
@@ -59,10 +60,10 @@
     Object[] EMPTY_ARRAY = new Object[] {};
     @SuppressWarnings("checkstyle:linelength")
     String SQL_FIND_ALL_CONFIG_INFO = "SELECT id,data_id,group_id,tenant_id,app_name,content,type,md5,gmt_create,gmt_modified,src_user,src_ip,c_desc,c_use,effect,c_schema,encrypted_data_key FROM config_info";
-
+
```
Comment (li-xiao-shuang) [inline config/src/main/java/com/alibaba/nacos/config/server/service/repository/PersistService.java:63]: 只提交修改的地方  不要格式化，可以安装下 checkstyle

**Review by li-xiao-shuang — APPROVED** [review]

**Review by KomachiSion — CHANGES_REQUESTED** [review]
Bad implementation.

auth is a plugin. but namespace controller is a basic featrue. So basic feature force dependent plugin implementation is not right.

**Comment by KomachiSion** [comment]
I think we need discuss in issue first. The implementation is quite unreasonable. close first.

**Comment by zrfan2016** [comment]
> Bad implementation.
> 
> auth is a plugin. but namespace controller is a basic featrue. So basic feature force dependent plugin implementation is not right.

when server restart , and token is available ，then can't get user from session. If ignore this case ,just get user from session,how about  ?

**Pull request closed without merging**